[stdlib] Arith: deprecate files mentioned as (mostly) obsolete (#14736)
coq/coq: doc/changelog/10-standard-library/14736-Arith2.rst

- **Deprecated:**

>   some obsolete files from the ``Arith`` part of the standard library (``Div2``, ``Even``, ``Gt``, ``Le``, ``Lt``, ``Max``, ``Min``, ``Minus``, ``Mult``, ``NPeano``, ``Plus``).

Reviewer: ```suggestion
  some obsolete files from the ``Arith`` part of the standard library (``Div2``, ``Even``,
  ``Gt``, ``Le``, ``Lt``, ``Max``, ``Min``, ``Minus``, ``Mult``, ``NPeano``, ``Plus``).
```

  Import ``Arith_base`` instead of these files.  References to items in the deprecated files should be replaced

>   with references to `PeanoNat` as suggested by the warning messages.

Author: ```suggestion
  with references to ``PeanoNat.Nat`` as suggested by the warning messages.
```

  All ``Hint`` declarations in the ``arith`` database have been moved to ``Arith_prebase`` and
  ``Arith_base``.  To use the results about Peano arithmetic, we recommend importing
  ``PeanoNat`` (or ``Arith_base`` to base it on the ``arith`` hint database) and using the ``Nat`` module.
  ``Arith_prebase`` has been introduced temporarily to ensure compatibility, but it will be removed at the end of the

>   deprecation phase, e.g. in 8.17.  Its use is thus discouraged.```

Reviewer: ```suggestion
  deprecation phase, e.g. in 8.17.  Its use is thus discouraged.
```

- **Deprecated:**
  some obsolete files from the ``Arith`` part of the standard library (``Div2``, ``Even``, ``Gt``, ``Le``, ``Lt``, ``Max``, ``Min``, ``Minus``, ``Mult``, ``NPeano``, ``Plus``).

>   ``Arith_prebase`` has been introduced temporarily to ensure compatibility, but it will be removed at the end of the
>   deprecation phase, e.g. in 8.17. Usage is thus discouraged.
>   All ``Hint`` declarations in the ``arith`` database have been moved to ``Arith_prebase`` and
>   ``Arith_base``.  To use the results about Peano arithmetic, we recommend importing
>   ``PeanoNat`` (or ``Arith_base`` to base it on the ``arith`` hint database) and using the ``Nat`` module.
>   To avoid deprecation warnings, ``Arith_base`` should be used in replacement of the deprecated files,
>   and calls to the statements from deprecated files should be replaced by the corresponding statements in ``PeanoNat.Nat``
>   suggested by the warning messages.

Reviewer: That's better.  How about this if I've not changed the meaning?

(Also I moved what you **_should_** do before what you **_shouldn't_** do)

```suggestion
  Import ``Arith_base`` instead of these files.  References to items in the deprecated files should be replaced
  with references to `PeanoNat` as suggested by the warning messages.
  All ``Hint`` declarations in the ``arith`` database have been moved to ``Arith_prebase`` and
  ``Arith_base``.  To use the results about Peano arithmetic, we recommend importing
  ``PeanoNat`` (or ``Arith_base`` to base it on the ``arith`` hint database) and using the ``Nat`` module.
  ``Arith_prebase`` has been introduced temporarily to ensure compatibility, but it will be removed at the end of the
  deprecation phase, e.g. in 8.17.  Its use is thus discouraged.```

- **Deprecated:**

>   Some obsolete files from the ``Arith`` part of the standard library are now explicitely deprecated (``Div2``, ``Even``, ``Gt``, ``Le``, ``Lt``, ``Max``, ``Min``, ``Minus``, ``Mult``, ``NPeano``, ``Plus``).

Reviewer 2: ```suggestion
  some obsolete files from the ``Arith`` part of the standard library (``Div2``, ``Even``, ``Gt``, ``Le``, ``Lt``, ``Max``, ``Min``, ``Minus``, ``Mult``, ``NPeano``, ``Plus``).
```

>   A temporary file ``Arith_prebase`` is introduced to ensure compatibility but should not be imported directly (it will disappear at the end of the deprecation phase).

Reviewer 2: ```suggestion
  A temporary file ``Arith_prebase`` has been introduced to ensure compatibility but should not be imported directly (it will disappear at the end of the deprecation phase).
```

Reviewer 2: What do you mean by "should not be imported directly"? Perhaps it might be better to say something like:
```suggestion
  A temporary file ``Arith_prebase`` has been introduced to ensure compatibility. Usage is discouraged since it will disappear at the end of the deprecation phase.
```

Reviewer: Maybe this?

```suggestion
  ``Arith_prebase`` has been introduced temporarily to ensure compatibility, but it will be removed at the end of the 
  deprecation phase, e.g. in 8.16.
```

Would you add a description of the actions users need to take as a result of this PR?   Seems like it's replace references to the deprecated files with (exactly what?).  And don't use Arith_prebase.  It's a bit odd to draw attention to `Arith_prebase` when we don't want users to import it directly.

Reviewer: BTW "explicitely" should be "explicitly"

Author: I have tried to give a description of what should be done to avoid deprecation warnings.
Shouldn't the end of deprecation phase be deprecation + 2? thus 8.17 if introduction in 8.15?
Anyway @Alizter and @jfehrle, thanks for your suggestions.

Reviewer 2: I think it's more likely that this PR will get into 8.16 so the EOF should be 8.18.

>   All ``Hint`` declarations in the ``arith`` database are moved to files ``Arith_prebase`` and ``Arith_base``.
>   In order to use the results about Peano arithmetic provided by the standard library, it is recommended to import ``PeanoNat`` (or ``Arith_base`` to take benefits from the ``arith`` hint database) and to use the ``Nat`` module
>   (`#14736 <https://github.com/coq/coq/pull/14736>`_,
>   by Olivier Laurent).

Reviewer 2: ```suggestion
  All ``Hint`` declarations in the ``arith`` database have been moved to files ``Arith_prebase`` and ``Arith_base``.
  In order to use the results about Peano arithmetic provided by the standard library, we recommend the import of ``PeanoNat`` (or ``Arith_base`` to build on top of the ``arith`` hint database) and to use the ``Nat`` module
  (`#14736 <https://github.com/coq/coq/pull/14736>`_,
  by Olivier Laurent).
```

Reviewer: How about:

```suggestion
  All ``Hint`` declarations in the ``arith`` database have been moved to ``Arith_prebase`` and
  ``Arith_base``.  To use the results about Peano arithmetic, we recommend importing
  ``PeanoNat`` (or ``Arith_base`` to base it on the ``arith`` hint database) and using the ``Nat`` module
  (`#14736 <https://github.com/coq/coq/pull/14736>`_,
  by Olivier Laurent).
```
